src/openrct2/core/FlagHolder.hpp

```diff
@@ -0,0 +1,51 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+template<typename THolderType, typename TEnumType> struct FlagHolder
```

Comment:
Instead of specifying two template parameters, you can almost definitely leverage [std::underlying_type](https://en.cppreference.com/w/cpp/types/underlying_type).

Reply:
I know, but an enum might be used in multiple places, e.g. RideMode has an underlying type of `uint8_t` and is saved as such with a ride, but when using it as a flag you need a bigger integer.